docs/contributors/testing.md

```diff
@@ -21,26 +21,36 @@ Used:
 
 Tips
 - Try to always add such tests, they are fast, they cover a lot of use cases as they run at low level 
-- They may be hard to apply to mxgraph integration code especially when interacting with the mxGraph API (prefer e2e tests in that case)
+- They may be hard to apply to mxgraph integration code especially when interacting with the mxGraph API (prefer integration or at last end-to-end tests in that case)
 - To facilitate _json to model_ tests there is a CLI tool to generate json and/or internal model from bpmn file - please refer to [CLI parseBpmn](../../scripts/utils/README.md)
 - You can have a look at the existing tests for inspiration, the good example would be: [BpmnJsonParser.sequenceFlow](../../test/unit/component/parser/json/BpmnJsonParser.sequenceFlow.normal.test.ts)
 
-### End to end tests
+
+### Integration tests
+
+We have the following types of integration tests, to check the integration with `mxGraph`:
+- model: check the library fills the `mxGraph` model with BPMN elements. The model must contain the related vertices and
+  edges, with right coordinates, dimensions and style.
+- dom: `JSDOM` based, no browser, check SVG elements are created or updated accordingly. They are used for both non regression
+(`mxGraph` integration) and DOM additions for custom interactions.
+
+Tests change frequency:
+- The `dom` tests are very rarely changed, as they are testing the `mxGraph` integration at very high level.
+- The model tests must be updated each time the library supports a new BPMN element.
+
+
+### End-to-end tests
 
 We use the [puppeteer library](https://github.com/puppeteer/puppeteer/) with the [puppeteer jest integration](https://jestjs.io/docs/en/puppeteer) to handle
 tests requiring a web browser (Chromium only for now).
 
-We have the following types of end to end tests:
-- model: check the library fills the `mxGraph` model with BPMN elements. The model must contain the related vertices and
-edges, with right coordinates, dimensions and style.
-- dom: `JSDOM` based, no browser, check SVG nodes as in view tests
-- view: ensure SVG elements are present on the web page (mainly non regression, to verify that the `mxGraph` integration uses SVG for rendering)
+We have the following types of end-to-end tests:
+- `generated svg`: ensure SVG elements are present on the web page (mainly non regression, to verify that the `mxGraph` integration uses SVG for rendering)
```

Comment:
I don't understand the difference with the visual tests 🙁

Reply:
`ensure SVG elements are present on the web page` --> `ensure SVG elements are present in the DOM of the web page` would be clearer? the text hasn't changed with this PR

Reply:
Maybe we can speak drirectly. But I have the feeling that the Generated SVG test are the same as the Visual tests.

Reply:
yes let's talk about it.
They are not the same. In ' Generated SVG' tests, we check the DOM and search for svg element. Historically, this was the first end-to-end tests we have introduced. They are very close to the 'DOM' integration tests except that we check the DOM in the browser here instead of with JSDOM in integration tests.
In visual tests, we are focussing on the actual rendering, what the user sees.

Reply:
It's ok for me if you add your proposition 😉

Reply:
done in c02c39f